exasol_sphinx_github_pages_generator/deploy_github_pages.py

```diff
@@ -153,25 +18,22 @@ def deploy_github_pages(argv):
     print("PUSH_ENABLED=" + args.push_enabled)
     print("SOURCE_BRANCH=" + args.source_branch + "\n")
 
-    current_commit_id = run(["git", "rev-parse", "HEAD"], capture_output=True, text=True, check=True)
     with TemporaryDirectory() as tempdir:
-        worktree = tempdir + "/worktree"
-        build_dir = tempdir + "/build"
-        intermediate_dir = tempdir + "/intermediate"
-        os.mkdir(build_dir)
-
+        deployer = GithubPagesDeployer(source_dir, args.module_path, args.target_branch,
+                                       args.push_origin, args.push_enabled, args.source_branch,
+                                       current_commit_id, tempdir)
+        os.mkdir(deployer.build_dir)
         print("Using following Directories:")
         print("TMP=" + tempdir)
```

Comment:
I personally think that looks more confusing in the code, because you have a list of names and the a list of parameters and have to parse them together in your head. Is there some other benefit to this that i am overlooking?
Changing the single prints to [formatted string literals/f-strings](https://docs.python.org/3/reference/lexical_analysis.html#f-strings) might be better though.

Reply:
@Nicoretti out of curiosity what does the cleandoc do?

Reply:
@MarleneKress79789 you could also do
```python 
cleandoc(f"""
.....
""")
```

@tkilias basically it is what python itself uses to clean the  `__doc__` doc attributes:

> Clean up indentation from docstrings that are indented to line up with blocks of code.
> 
> All leading whitespace is removed from the first line. Any leading whitespace that can be uniformly removed from the second line onwards is removed. Empty lines at the beginning and end are subsequently removed. Also, all tabs are expanded to spaces.

Reply:
@Nicoretti thx, nice function.

Reply:
done